Sources/Clients/SecureStorageClient/SecureStorageClient+Interface.swift

```diff
@@ -37,6 +38,7 @@ extension SecureStorageClient {
 	public typealias SaveProfileHeaderList = @Sendable (ProfileSnapshot.HeaderList) async throws -> Void
 	public typealias DeleteProfileHeaderList = @Sendable () async throws -> Void
 
+	public typealias GetDeviceIdentifierSetIfNil = @Sendable (UUID) async throws -> UUID
```

Comment:
I am not really sure this is the best idea, this is both a Query(read the identifier) and a Command(save the identifier).
Additionally, shouldn't it be a function that composes `LoadDeviceIdentifier` and `SaveDeviceIdentifier`?

Reply:
I agree, I think we want to split up reading, generating and saving

Reply:
You are both missing one crucial detail. We CANNOT compose `Load` and `Save` because that happens outside of the actor, thus is not actor isolated, thus not data race free. Which I've asserted in the unit tests.....

So the actor must have this function and expose it.

This is used in `claimProfileSnapshot` which calls `saveDeviceIdentifierIfNeeded`, which before this PR was:
```swift
	public func saveDeviceIdentifierIfNeeded(_ deviceID: UUID) async throws {
		if let existing = try? await loadDeviceIdentifier() {
			if existing != deviceID {
				try await saveDeviceIdentifier(deviceID)
			}
		} else {
			try await saveDeviceIdentifier(deviceID)
		}
	}
```

which is NOT data race free, since it happens outside of the Keychain actor.

Reply:
But why should this be on SecureStorageClient actor? I mean `ProfileStore` is also an actor.

Reply:
@GhenadieVP SecureStorageClient is NOT an Actor. 

In code, ONLY SecureStorageClient dep should use KeychainClient, all other code always goes through SecureStorageClient.... it is an abstraction on top of Keychain.

Reply:
Ok, my bad missed that SecureStorageClient is not actor. Would this `GetDeviceIdentifierSetIfNil` be used outside of ProfileStore? If not, how is `GetDeviceIdentifierSetIfNil` on SecureStorageClient safer than `saveDeviceIdentifierIfNeeded` on ProfileStore?

Reply:
there is no "saveDeviceIdentifierIfNeeded on ProfileStore", before this PR ProfileStore uses SecureStorageClient - as it should - and after this PR ProfileStore still uses SecureStorage. And SecureStorage, is an abstraction layer on top of Keychain, as it has always been.

The change in this PR is that the bug we had in `main` before, with multiple competing `loadElseGenerateAndSave` is not possible, the bug is eradicated. Apart from we are not doing it like that anymore... since https://github.com/radixdlt/babylon-wallet-ios/pull/828
This PR does makes `claimProfile` even more safe! But it is not called from multiple tasks... but if it would, it would still be safe.

Reply:
Fair enough, I am still not sold on having this, unless you mean to have it as short term solution for the current state of the code. We do use that `saveDeviceIdentifierIfNeeded` as a workaround for the bug, and it really shouldn't be there.

Reply:
@GhenadieVP this PR really does not change the behaviour of ProfileStore at all, at least that is my claim! The only thing is that that function `saveDeviceIdentifierIfNeeded` has been made more safe. That is all.